scenarios/aws/ec2/vm_run.go

```diff
@@ -0,0 +1,110 @@
+package ec2
+
+import (
+	"github.com/DataDog/test-infra-definitions/components/datadog/agent"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/DataDog/test-infra-definitions/components/datadog/apps/dogstatsd"
+	"github.com/DataDog/test-infra-definitions/components/datadog/dockeragentparams"
+	"github.com/DataDog/test-infra-definitions/components/docker"
+	"github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/resources/aws"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/fakeintake"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func VMRun(ctx *pulumi.Context) error {
+	env, err := aws.NewEnvironment(ctx)
+	if err != nil {
+		return err
+	}
+
+	osDesc := os.DescriptorFromString(env.InfraOSDescriptor())
+	vm, err := NewVM(env, "vm", WithAMI(env.InfraOSImageID(), osDesc, osDesc.Architecture))
+	if err != nil {
+		return err
+	}
+	if err := vm.Export(ctx, nil); err != nil {
+		return err
+	}
+
+	if env.AgentDeploy() {
+		agentOptions := []agentparams.Option{}
+		if env.AgentUseFakeintake() {
+			fakeIntakeOptions := []fakeintake.Option{}
+
+			if env.InfraShouldDeployFakeintakeWithLB() {
+				fakeIntakeOptions = append(fakeIntakeOptions, fakeintake.WithLoadBalancer())
+			}
+
+			fakeintake, err := fakeintake.NewECSFargateInstance(env, vm.Name(), fakeIntakeOptions...)
+			if err != nil {
+				return err
+			}
+			agentOptions = append(agentOptions, agentparams.WithFakeintake(fakeintake))
+		}
+
+		_, err = agent.NewHostAgent(env.CommonEnvironment, vm, agentOptions...)
+		return err
+	}
+
+	return nil
+}
+
+func VMRunWithDocker(ctx *pulumi.Context) error {
+	env, err := aws.NewEnvironment(ctx)
+	if err != nil {
+		return err
+	}
+
+	osDesc := os.DescriptorFromString(env.InfraOSDescriptor())
+	vm, err := NewVM(env, "vm", WithAMI(env.InfraOSImageID(), osDesc, osDesc.Architecture))
```

Comment:
**🔨 warning**
We should limit to use with Amazon ECS, and avoid installing Docker. This is a breaking change compared to what we have today.

Reply:
I see that it was changed indeed https://github.com/DataDog/test-infra-definitions/pull/239.

I don't think we should limit test cases, especially for manual testing. We can handle in code to flip the boolean if the OS is `AmazonLinuxECS` but I see no valid reason to artificially limit Docker.

Reply:
I don't see usage of Docker out of `Amazon ECS`. For Windows I don't think the docker installer works today. So I don't see why we should not use the one with Docker out of the box. Maybe for testing on ne versions of Docker ?

Reply:
We have many use cases to test Docker with specific distribution, Kernel version, or cgroup version (v1 vs v2), to reproduce some customer cases or to manually test features (like cgroupv2 support recently).

Reply:
I see, we should make it more explicit that the recommended way is using it on Amazon ECS to have docker pre-installed. With this breaking change we are removing a limitation, we should replace it with explicit guidelines.